test/functional/mongo_client_tests.js

```diff
@@ -436,6 +436,20 @@ exports["correctly error out when no socket available on MongoClient.connect"] =
   }
 }
 
+exports["correctly handles compression"] = {
```

Comment:
I'm not sure this test is quite useful atm, unless you can validate that the internal state object was created with the expected compression options. I imagine you will have far more comprehensive tests soon that actually verify that the expected compression is used based on URI (_actually_ testing the compression etc)